pkg/controller/v1beta1/inferenceservice/reconcilers/service/service_reconciler.go

```diff
@@ -157,8 +157,7 @@ func (r *ServiceReconciler) checkServiceExist(client client.Client) (constants.C
 }
 
 func semanticServiceEquals(desired, existing *corev1.Service) bool {
-	return equality.Semantic.DeepEqual(desired.Spec.Ports, existing.Spec.Ports) &&
-		equality.Semantic.DeepEqual(desired.Spec.Selector, existing.Spec.Selector)
+	return equality.Semantic.DeepEqual(desired.Spec, existing.Spec)
```

Comment:
KServe does not use all the service spec fields, so there could be diffs generated by the Kubernetes defaults which causes infinite reconciliation loop.

Reply:
Good point. Closing this